### PR TITLE
feat: use cxg discover api to list datasets in a collection (#2337)

### LINKED ISSUE #2337: Use CELLxGENE Discover API to list datasets in a collection
and not the dp API

### PR DESCRIPTION
**Pull request by frano-m**
Closes #2337.

### CODE REVIEW

File: next/@types/network.ts [hunk 1]
```diff
@@ -107,10 +102,8 @@ export type IntegratedAtlasRow = AtlasRow &
   Pick<IntegratedAtlas, "cxgId" | "cxgURL" | "datasetAssets" | "name">;
 
 export interface DatasetAsset {
-  assetId: string;
-  datasetId: string;
   downloadURL: string;
-  fileName: string;
+  fileSize: number;
```
Comment (frano-m) [inline next/@types/network.ts:106]: `filesize` is returned from CELLxGENE dataset API - we can now display the download file size.

File: next/utils/atlasPages.ts [hunk 2]
```diff
@@ -62,10 +62,12 @@ export const getStaticProps: GetStaticProps<AtlasContext> = async (
   const cxgDatasets = [];
   if (atlas.cxgId) {
     const response = await fetch(
-      `https://api.cellxgene.cziscience.com/dp/v1/collections/${atlas.cxgId}`
+      `https://api.cellxgene.cziscience.com/curation/v1/collections/${atlas.cxgId}`
```
Comment (frano-m) [inline next/utils/atlasPages.ts:65]: API updated to CELLxGENE Discover API.

File: next/utils/atlasPages.ts [hunk 3]
```diff
@@ -62,10 +62,12 @@ export const getStaticProps: GetStaticProps<AtlasContext> = async (
...
     );
     const cxgCollection = await response.json();
-    cxgDatasets.push(...cxgCollection.datasets);
+    cxgDatasets.push(
+      ...mapDatasets(cxgCollection.datasets, cxgCollection.collection_id)
```
Comment (frano-m) [inline next/utils/atlasPages.ts:69]: `collection_id` is not in the response (datasets)...

File: next/utils/networkPages.ts [hunk 4]
```diff
@@ -35,7 +35,7 @@ export const getStaticProps: GetStaticProps<NetworkParam> = async (
   }
 
   const cxgResponse = await fetch(
-    "https://api.cellxgene.cziscience.com/dp/v1/datasets/index"
+    "https://api.cellxgene.cziscience.com/curation/v1/datasets"
```
Comment (frano-m) [inline next/utils/networkPages.ts:38]: API updated to CELLxGENE Discover API.

**Review by NoopDog — APPROVED** [review]
Thanks @frano-m ! ⭐ ⭐ !!

**Pull request merged**